x-pack/plugins/security_solution/server/endpoint/routes/actions/response_actions.ts

```diff
@@ -290,50 +292,77 @@ function responseActionRequestHandler<T extends EndpointActionDataParameterTypes
   const logger = endpointContext.logFactory.get('responseActionsHandler');
 
   return async (context, req, res) => {
+    logger.debug(`response action [${command}]:\n${dump(req.body)}`);
+
+    // Note:  because our API schemas are defined as module static variables (as opposed to a
+    //        `getter` function), we need to include this additional validation here, since
+    //        `agent_type` is included in the schema independent of the feature flag
+    if (
+      req.body.agent_type &&
+      !endpointContext.experimentalFeatures.responseActionsSentinelOneV1Enabled
+    ) {
+      return errorHandler(
+        logger,
+        res,
+        new CustomHttpRequestError(`[request body.agent_type]: feature is disabled`, 400)
+      );
+    }
+
     const user = endpointContext.service.security?.authc.getCurrentUser(req);
     const esClient = (await context.core).elasticsearch.client.asInternalUser;
     const casesClient = await endpointContext.service.getCasesClient(req);
-    const actionsClient = new EndpointActionsClient({
-      esClient,
-      casesClient,
-      endpointContext,
-      username: user?.username ?? 'unknown',
-    });
+    const connectorActions = (await context.actions).getActionsClient();
+    const responseActionsClient: ResponseActionsClient = getResponseActionsClient(
+      req.body.agent_type || 'endpoint',
+      {
+        esClient,
+        casesClient,
+        endpointService: endpointContext.service,
+        username: user?.username || 'unknown',
+        connectorActions,
+      }
+    );
 
     try {
       let action: ActionDetails;
 
       switch (command) {
         case 'isolate':
```

Comment:
I don't want to sound like a broken record :D but maybe we could drop hardcoded strings in favor of an enum? Keep all these in one place